app/controllers/notification.py

```diff
@@ -127,4 +134,47 @@ async def read_notification(
             notification,
             auto_commit=True,
         )
+        channels.publish(  # type: ignore
+            {"user_id": str(user.id), "id": str(notification.id), "event": "updated"},
+            "notification_events",
+        )
         return notifications_service.to_schema(notification, schema_type=schemas.Notification)
+
+    @websocket("/api/v1/users/{user_id:uuid}/notification/events/stream")
+    async def stream_notification_events(
+        self,
+        socket: WebSocket[Any, Any, Any],
+        channels: ChannelsPlugin,
+        notifications_service: NotificationService,
+        users_service: UserService,
+        user_id: uuid.UUID,
+        unread: bool | None = None,
+    ) -> None:
+        await socket.accept()
+
+        user: models.User | None = await users_service.get_one_or_none(id=user_id)
+        if user is None:
+            await socket.close()
+            return
+
+        async def _sender(message: str) -> None:
+            data = json.loads(message)
+            if data["user_id"] != str(user_id):
+                return
+            await socket.send_json(data)
+
+        # then listen for notification events
+        async with (
+            channels.start_subscription(["notification_events"]) as subscriber,
+            subscriber.run_in_background(_sender),  # type:ignore
+        ):
+            while True:
+                try:
+                    message = await socket.receive_text()
+                    if message == "close":
+                        # XXX this is for tests
+                        await socket.close()
+                        return
+                except WebSocketDisconnect:
+                    # if the socket is closed, avoid exception trace
+                    return
```

Comment:
I have a proposition, but I'm not sure it's really useful (at least for now that we never expect any messages being sent by the frontend on the websocket):

```suggestion
            while True:
                try:
                    # Use a timeout to prevent blocking indefinitely
                    # This allows the background _sender task to process channel messages
                    message = await asyncio.wait_for(socket.receive_text(), timeout=0.1)
                    if message == "close":
                        # Close connection if client sends "close" message
                        await socket.close()
                        return
                except asyncio.TimeoutError:
                    # Timeout is expected, continue listening for channel messages
                    continue
                    message = await socket.receive_text()
                except WebSocketDisconnect:
                    # if the socket is closed, avoid exception trace
                    return

```